joint_position_controller:
    type: franka_controllers/JointPositionController
    arm_id: $(arg arm_id)
    joint_names:
        - $(arg arm_id)_joint1
        - $(arg arm_id)_joint2
        - $(arg arm_id)_joint3
        - $(arg arm_id)_joint4
        - $(arg arm_id)_joint5
        - $(arg arm_id)_joint6
        - $(arg arm_id)_joint7

franka_state_controller:
  type: franka_control/FrankaStateController
  publish_rate: 30  # [Hz]
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7

# LCQP_controller_standalone:
#   type: franka_controllers/LCQPControllerStandalone
#   arm_id: $(arg arm_id)
#   joint_names:
#     - $(arg arm_id)_joint1
#     - $(arg arm_id)_joint2
#     - $(arg arm_id)_joint3
#     - $(arg arm_id)_joint4
#     - $(arg arm_id)_joint5
#     - $(arg arm_id)_joint6
#     - $(arg arm_id)_joint7
#   goal: [0.5, 0.4, 0.6]
#   obstaclePosition: [0.47, 0.07, -0.55]
#   obstacleRadius: 0.1
#   constMainTask: 1
#   constContactTask: 0.001
#   solverBound: 100
#   safetyDistance: 0.15
#   linkObstacleAvoid: [2, 3, 4, 6]
#   robustJinvLambda: 0.001
#   maximumTime: 40.0
#   tolerance: 0.01

LCQP_controller_frankaModel:
  type: franka_controllers/LCQPControllerFrankaModel
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7
  goal: [0.1, 0.450, 0.5]
  obstaclePosition: [0.29, 0.27, 0.54]
  obstacleRadius: 0.1
  constMainTask: 0.1    # 0.5
  constContactTask: 0.05
  constError: 0.2
  solverBound: 100
  safetyDistance: 0.15
  linkObstacleAvoid: [2, 3, 4, 6]
  robustJinvLambda: 0.001
  maximumTime: 40.0
  tolerance: 0.01
  LowPassJoint: 500
  LowPassObs: 800

LCQP_controller_multipleObstacle:
  type: franka_controllers/LCQPControllerMultipleObstacle
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7
  goal: [0.5, 0.4, 0.3]
  obstacleRadius: 0.1
  constMainTask: 0.2    # 0.5
  constContactTask: 0.05
  constError: 0.2
  solverBound: 100
  safetyDistance: 0.15
  safetyIgnore: 0.17
  linkObstacleAvoid: [2, 3, 6]
  robustJinvLambda: 0.001
  maximumTime: 40.0
  tolerance: 0.01
  jacobianSkipcount: 1
